VMT/KCSA/KCSA-CVE-2020-2023.md

```diff
@@ -0,0 +1,57 @@
+announcement-date: 2020-06-12
+
+id: KCSA-CVE-2020-2023
+
+title: Kata Containers Execution with Unnecessary Privileges
+
+description: A container can access the guest root file system device.
+This can be used to gain code execution on the guest and masquerade as kata-agent.
+
+affected-components:
+
+  - components: `kata-agent`
+    version: Before v1.11.1
+
+vulnerabilities:
+
+  - CVE-ID: CVE-2020-2023
+
+reporters:
+
+  - name: `Yuval Avrahami`
+    affiliation: `Palo Alto Networks`
+    reported:
+      - CVE-2020-2023
+
+issues:
+
+  links:
+    - https://github.com/kata-containers/agent/issues/791
+
+ reviews:
+
+  v1.11.1:
+    - https://github.com/kata-containers/agent/pull/792
+  
+  type: GitHub
+
+reproduce:
+
+  - When Kata Containers is run with nvdimm rootfs image, a malicious guest can 
+    create a device file for the guest root filesystem device using mknod, and gain access to the filesystem as:  
+    1. List `/sys/dev/block` path and get guest root filesystem device major and minor numbers.
+    2. Perform a `mknod on` the guest major and minor numbers.
+    3. Delete the kata-agent binary on the guest.
+    4. Write several big files with garbage data to the device so that the blocks 
+       previously allocated for the kata-agent binary will be used.
+    5. In case of DAX, the kata-agent process code region is directly mapped to 
+       the device and now points to garbage data, resulting in a `SIGSEGV`.
+    6. In case of no DAX, similar outcome can be achieved by running a process to 
+       exhaust memory and cause guest kernel to free page cache including pages allocated
+       for the kata-agent binary. The kernel would need to remap the code region 
+       of the kata-agent process to its executable, and read the kata-agent 
+       binary directly from the compromised guest filesystem device. 
+
+notes:
+  - The vulnerability can be used to compromise the guest and masquerade 
+    as the kata-agent. All users are recommended to upgrade to mitigate guest breakout.
```

Comment:
Consider adding:
'
To exploit the issue, the container must possess CAP_MKNOD.
'